examples/configurable_chain/server.py

```diff
@@ -38,26 +59,46 @@
 )
 chain = prompt | model | StrOutputParser()
 
-app = FastAPI(
-    title="LangChain Server",
-    version="1.0",
-    description="Spin up a simple api server using Langchain's Runnable interfaces",
-)
+# Add routes requires you to specify which config keys are accepted
+# specifically, you must accept `configurable` as a config key.
+add_routes(app, chain, path="/config_from_rc", config_keys=["configurable"])
 
-# Set all CORS enabled origins
-app.add_middleware(
-    CORSMiddleware,
-    allow_origins=["*"],
-    allow_credentials=True,
-    allow_methods=["*"],
-    allow_headers=["*"],
-    expose_headers=["*"],
+
+###############################################################################
+#             EXAMPLE 2: Configure fields based on Request metadata           #
+###############################################################################
+
+
+# Add another example route where you can configure the model based
+# on properties of the request. This is useful for passing in API
+# keys from request headers (WITH CAUTION) or using other properties
+# of the request to configure the model.
+def fetch_api_key_from_header(config: Dict[str, Any], req: Request) -> Dict[str, Any]:
+    if "x-api-key" in req.headers:
+        config["configurable"]["openai_api_key"] = req.headers["x-api-key"]
```

Comment:
1) Do you know what happens in terms of tracing if tracing is ON? i.e., do the secrets end up leaking to the metadata?
2) I suspect a more common use case will be to validate the user identity, and then map something about user identity to some information via databases models; anything that would prevent a developer using langserve from being able to do that? If not, maybe we could add a comment explaining that this logic can happen here as well? Does any of this works weirdly with Depends patterns in fast api?

Reply:
For 1, I tested and it looks like the value gets piped into a secret field (the openAI secret), so I didn't see anything in the trace when I looked through on langsmith

For 2, can you give me an example of what sort of Depends pattern may be affected here / intuition on why you think this might be an issue?